haystack/nodes/image_to_text/base.py

```diff
@@ -0,0 +1,55 @@
+from typing import List, Optional
+
+from abc import abstractmethod
+
+from haystack.schema import Document
+from haystack.nodes.base import BaseComponent
+
+
+class BaseImageToText(BaseComponent):
+    """
+    Abstract class for ImageToText
+    """
+
+    outgoing_edges = 1
+
+    @abstractmethod
+    def generate_captions(
+        self, image_file_paths: List[str], generate_kwargs: Optional[dict] = None, batch_size: Optional[int] = None
+    ) -> List[Document]:
+        """
+        Abstract method for generating captions.
+
+        :param image_file_paths: Paths of the images
+        :param generate_kwargs: Dictionary containing arguments for the generate method of the Hugging Face model.
+                                See https://huggingface.co/docs/transformers/en/main_classes/text_generation#transformers.GenerationMixin.generate
+        :param batch_size: Number of images to process at a time.
+        :return: List of Documents. Document.content is the caption. Document.meta["image_file_path"] contains the image file path.
+        """
+        pass
+
+    def run(self, file_paths: Optional[List[str]] = None, documents: Optional[List[Document]] = None):  # type: ignore
+
+        if file_paths is None and documents is None:
+            raise ValueError("You must either specify documents or image file_paths to process.")
+        if file_paths is not None and documents is not None:
+            raise ValueError(
+                "You specified both documents and image_file_paths. You need to specify only one of the two parameters."
+            )
```

Comment:
I can't imagine when it would happen that a node receives both these values, but even if it does, I don't think it's an issue, right? We can simply merge the paths from the documents list to the file_paths into `image_file_paths`